src/backend/cpu/device_manager.cpp

```diff
@@ -138,8 +138,22 @@ DeviceManager& DeviceManager::getInstance() {
 
 CPUInfo DeviceManager::getCPUInfo() const { return cinfo; }
 
+void DeviceManager::resetMemoryManager() {
+    // Destroys the existing memory manager, but does not release the
+    // corresponding handle, if there was one
```

Comment:
Where should we handle the release of existing handle in that case ?

Reply:
I'm trying to think about the best way to do this now. I don't think we should release the handle, and I actually don't think we should call `shutdown` on the memory manager either - it's conceivable that you would want to swap back and forth between memory managers while still maintaining all of their internal state.

@umar456 @9prady9 thoughts on the best behavior here?